quinn-proto/src/connection/streams/recv.rs

```diff
@@ -287,10 +287,7 @@ impl<'a> Chunks<'a> {
                     self.state = ChunksState::Finished;
                     Ok(None)
                 } else {
-                    let should_transmit =
-                        Self::done(&mut rs, self.read, self.id, self.streams, self.pending);
-                    self.state = ChunksState::Error(ReadError::Blocked, should_transmit);
-                    self.streams.recv.insert(self.id, rs);
+                    self.state = ChunksState::Readable(rs);
```

Comment:
It feels a bit counterintuitive that we just set `state` back to `Readable` even though we're returning `Blocked` here. Maybe add a comment with some context?

Reply:
Added some discussion.